plugins/modules/route53_ksk.py

```diff
@@ -0,0 +1,367 @@
+
+
+def get_hosted_zone(client, hosted_zone_id):
+    return client.get_hosted_zone(Id=hosted_zone_id)
+
+
+def get_dnssec(client, hosted_zone_id):
+    return client.get_dnssec(HostedZoneId=hosted_zone_id)
+
+
+def find_ksk(client, module):
+    hosted_zone_dnssec = get_dnssec(client, module.params.get("hosted_zone_id"))
+    if hosted_zone_dnssec["KeySigningKeys"] != []:
+        for ksk in hosted_zone_dnssec["KeySigningKeys"]:
+            if ksk["Name"] == module.params.get("name"):
+                return ksk
+    return {}
+
+
+def wait(client, module, change_id):
+    try:
+        waiter = get_waiter(client, "resource_record_sets_changed")
+        waiter.wait(
+            Id=change_id,
+            WaiterConfig=dict(
+                Delay=5,
+                MaxAttempts=module.params.get("wait_timeout") // 5,
+            ),
+        )
+    except botocore.exceptions.WaiterError as e:
+        module.fail_json_aws(e, msg="Timeout waiting for changes to be applied")
+
+
+def create_or_update(client, module: AnsibleAWSModule, ksk):
+    changed: bool = False
+    zone_id = module.params.get("hosted_zone_id")
+    name = module.params.get("name")
+    status = module.params.get("status")
+    response = {}
+
+    if ksk:
+        if ksk["Status"] != status:
+            changed = True
+
+            if module.check_mode:
+                ksk["Status"] = status
+                module.exit_json(
+                    changed=changed,
+                    key_signing_key=camel_dict_to_snake_dict(ksk),
+                    msg=f"Would have updated the Key Signing Key status to {status} if not in check_mode.",
+                )
+
+            if status == "ACTIVE":
+                response.update(activate(client, zone_id, name))
+            elif status == "INACTIVE":
+                response.update(deactivate(client, zone_id, name))
+    else:
+        changed = True
+        if module.check_mode:
+            module.exit_json(changed=changed, msg="Would have created the Key Signing Key if not in check_mode.")
+
+        response = client.create_key_signing_key(
+            CallerReference=module.params.get("caller_reference"),
+            KeyManagementServiceArn=module.params.get("key_management_service_arn"),
+            HostedZoneId=zone_id,
+            Name=name,
+            Status=status,
+        )
+
+    return changed, response
+
+
+def delete(client, module: AnsibleAWSModule, ksk):
+    changed: bool = False
+    zone_id = module.params.get("hosted_zone_id")
+    name = module.params.get("name")
+    response = {}
+
+    if ksk:
+        changed = True
+        if module.check_mode:
+            module.exit_json(
+                changed=changed,
+                key_signing_key={},
+                msg="Would have deleted the Key Signing Key if not in check_mode.",
+            )
+
+        result = deactivate(client, zone_id, name)
+        change_id = result["ChangeInfo"]["Id"]
+        wait(client, module, change_id)
```

Comment:
```suggestion
```
The ``wait`` operation is performed on line ``351``, this one is not conditional

Reply:
The wait is necessary to wait for the deactivation of the key signing key, which must be done before deleting it. The wait operation on line 351 waits for deletion when wait=true.